packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -50,29 +54,31 @@ async function proxyPreviewModeRouteHandler(
   _context: Context,
   { apiKey }: { apiKey: string },
 ): Promise<NextResponse<ProxyPreviewModeResponse>> {
-  const secret = request.nextUrl.searchParams.get('x-makeswift-preview-mode')
+  const secret = request.nextUrl.searchParams.get('x-makeswift-draft-mode')
 
   if (secret !== apiKey) return new NextResponse('Unauthorized', { status: 401 })
 
+  const draftModeData: MakeswiftDraftData = {
+    makeswift: true,
+    siteVersion: MakeswiftSiteVersion.Working,
+  }
+
+  draftMode().enable()
+
   const proxyUrl = request.nextUrl.clone()
-  proxyUrl.searchParams.delete('x-makeswift-preview-mode')
+  proxyUrl.searchParams.delete('x-makeswift-draft-mode')
 
   const proxyRequest = new NextRequest(proxyUrl, { headers: request.headers })
-  proxyRequest.headers.delete('x-makeswift-preview-mode')
+  proxyRequest.headers.delete('x-makeswift-draft-mode')
 
-  draftMode().enable()
   const draftModeCookie = cookies().get('__prerender_bypass')
   if (draftModeCookie) {
     proxyRequest.cookies.set(draftModeCookie)
-
-    const previewData: MakeswiftPreviewData = {
-      makeswift: true,
-      siteVersion: MakeswiftSiteVersion.Working,
-    }
-
-    proxyRequest.cookies.set('x-makeswift-draft-mode-data', JSON.stringify(previewData))
+    proxyRequest.cookies.set(MAKESWIFT_DRAFT_MODE_DATA_COOKIE, JSON.stringify(draftModeData))
   }
-  draftMode().disable()
+
+  // TODO: Delete Draft Mode cookie.
+  cookies().delete('__prerender_bypass')
```

Comment:
Is this out best option?

Also, should we remove the TODO?

Reply:
Is `.disable()` our best option? Specifically, are we worried about sending back a `Set-Cookie` header to the client that removes `__prerender_bypass`? Do we have other alternatives?

Reply:
Had in-person discussion: no better alternatives that we know of.